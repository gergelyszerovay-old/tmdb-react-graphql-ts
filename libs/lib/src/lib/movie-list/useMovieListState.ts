import {useApolloClient, useQuery, useReactiveVar} from "@apollo/client";
import {movieListTermVar, selectedMovieIdVar, similarMoviesListVar, useUpdateFetchState} from "../state-gql/cache";
import {QUERY_MOVIE_LIST} from "../state-gql/gql";
import useLazyQueryMovieDetails from "../state-gql/useLazyQueryMovieDetails";

export interface MovieListState {
  onMovieSelected: (any) => void;
  movieList: any;
  movieListError: string;
}

const useMovieListState = (): MovieListState => {
  const client = useApolloClient();

  const similarMoviesList = useReactiveVar(similarMoviesListVar);
  const movieListTerm = useReactiveVar(movieListTermVar);

  const {data: tmdbLoadMoviesData, error: tmdbLoadMoviesError, loading: tmdbLoadMoviesLoading} = useQuery(QUERY_MOVIE_LIST, {
    skip: movieListTerm === '',
    variables: {
      term: movieListTerm
    },

    // https://github.com/apollographql/apollo-client/issues/6759 or https://github.com/apollographql/react-apollo/issues/3717
    fetchPolicy: "network-only"
  });

  useUpdateFetchState('QUERY_MOVIE_LIST', {
    errorMessage: tmdbLoadMoviesError?.message ? tmdbLoadMoviesError.message : '',
    loading: tmdbLoadMoviesLoading
  });

  const [loadMovieDetails] = useLazyQueryMovieDetails(client, 'http://' + window.location.hostname + ':4200/wikipedia/');

  return {
    onMovieSelected: (node) => {
      selectedMovieIdVar(node.id);
      loadMovieDetails(node.id);
    },
    movieList: similarMoviesList ? similarMoviesList : tmdbLoadMoviesData?.search?.edges,
    movieListError: tmdbLoadMoviesError ? tmdbLoadMoviesError.message : '',
  };
};

export default useMovieListState;
