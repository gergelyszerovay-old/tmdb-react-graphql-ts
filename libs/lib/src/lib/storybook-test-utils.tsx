import React from 'react';
import {StylesProvider} from '@material-ui/styles';

// https://github.com/mui-org/material-ui/issues/9492
export const WithStableMuiClassnames = (c) => {
  const generateClassName = () => {
    let counter = 0
    return (rule, styleSheet) => `${styleSheet.options.classNamePrefix}-${rule.key}-${counter++}`
  }

  return (
    <StylesProvider generateClassName={generateClassName()}>
      {c}
    </StylesProvider>
  )
}
