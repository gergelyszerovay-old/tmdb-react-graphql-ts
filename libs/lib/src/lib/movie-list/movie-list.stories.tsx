import React from 'react';
import {MovieList} from './movie-list';
// @ts-ignore
import movieList from "./fixtures/movieList.json";
import {action} from "@storybook/addon-actions";
import {MovieListState} from "./useMovieListState";

export default {
  component: MovieList,
  title: 'MovieList',
};

export const primary = () => {
  const state = (): MovieListState => {
    return {
      onMovieSelected: action('onMovieSelected'),
      movieListError: '',
      movieList
    }
  };
  return <MovieList hook={state}/>;
};

export const error = () => {
  const state = (): MovieListState => {
    return {
      onMovieSelected: action('onMovieSelected'),
      movieListError: 'Error',
      movieList: undefined
    }
  };
  return <MovieList hook={state}/>;
};
