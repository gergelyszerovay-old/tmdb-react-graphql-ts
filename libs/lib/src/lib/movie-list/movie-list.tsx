import React, {FC} from 'react';

import './movie-list.scss';
import {makeStyles} from "@material-ui/core/styles";
import {Chip, Divider, List, ListItem, ListItemText} from "@material-ui/core";
import useMovieListState, {MovieListState} from "./useMovieListState";
import MuiAlert from "@material-ui/lab/Alert";

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
    '& .MuiListItemText-primary': {},
    '& .MuiListItem-root': {
      cursor: 'pointer',
      display: 'block'
    },
    '& .MuiChip-root': {
      cursor: 'pointer',
    }
  },
  genreContainer: {
    '& > div': {
      marginRight: '5px',
      marginBottom: '5px'
    }
  },
  error: {
    margin: '10px'
  }
}));


/* eslint-disable-next-line */
export interface MovieListProps {
  hook?: () => MovieListState;
}

export const MovieList: FC<MovieListProps> = ({hook = useMovieListState}: MovieListProps) => {
  const classes = useStyles();

  const {movieList, onMovieSelected, movieListError} = hook();

  if (movieListError !== '') {
    return <MuiAlert elevation={6} variant="filled" severity="error" className={classes.error}>Can't load the movie list
      from the server!</MuiAlert>
  }

  if (movieList) {
    const items = movieList.map((edge: any, i: number) => {
      if (edge.node.__typename !== 'MovieResult' && edge.node.__typename !== '__Movie') {
        return (<React.Fragment key={'empty' + i}/>)
      }
      const genres = edge.node.details.genres.map((genre: any) => {
        return (
          <Chip size="small" label={genre.name} key={edge.node.id + '-' + genre.id}/>
        );
      });
      return (
        <React.Fragment key={edge.node.id}>
          <ListItem
            button
            onClick={() => {
              onMovieSelected(edge.node)
            }}
            data-testid={'movie-' + edge.node.id}
          >
            <ListItemText
              primary={edge.node.title + ` (${edge.node.rating})`}
            />
            <div className={classes.genreContainer}>{genres}</div>
          </ListItem>
          <Divider component="li"/>
        </React.Fragment>
      )
    });
    return (
      <List className={classes.root}>
        {items}
      </List>
    );

  }
  return <p/>
};

export default MovieList;
