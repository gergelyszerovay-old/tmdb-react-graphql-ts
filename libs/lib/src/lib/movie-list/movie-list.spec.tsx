import React from 'react';
import {fireEvent, getByTestId, render} from '@testing-library/react';

import MovieList from './movie-list';
import * as Stories from "./movie-list.stories";
import {WithStableMuiClassnames} from "../storybook-test-utils";
// @ts-ignore
import movieList from "./fixtures/movieList.json";
import {MovieListState} from "./useMovieListState";

describe('MovieList', () => {
  describe('On movie list item selectiom', () => {
    let container = null;
    let onMovieSelected = null;

    beforeEach(() => {
      onMovieSelected = jest.fn();
      const useMockState = (): MovieListState => {
        return {
          onMovieSelected: onMovieSelected,
          movieListError: '',
          movieList
        };
      };

      container = render(<MovieList hook={useMockState}/>).container;
    })

    it('Click on list item should trigger an event', () => {
      const buttonEl = getByTestId(container, 'movie-126889');
      fireEvent.click(buttonEl);
      expect(onMovieSelected).toHaveBeenCalledWith(
        expect.objectContaining({
          id: 126889
        })
      );
    });
  });


  describe('Storybook stories snapshot tests', () => {
    for (const [key, value] of Object.entries(Stories)) {
      if (value instanceof Function) {
        it(`Story ${key}`, () => {
          const {container} = render(WithStableMuiClassnames(Stories[key]()));
          expect(container.firstChild).toMatchSnapshot()
        });
      }
    }
  });
});
