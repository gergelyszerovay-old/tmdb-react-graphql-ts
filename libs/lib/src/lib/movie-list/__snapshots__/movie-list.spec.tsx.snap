// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList Storybook stories snapshot tests Story error 1`] = `
<div
  class="MuiPaper-root-19 MuiAlert-root-3 MuiAlert-filledError-15 makeStyles-error-2 MuiPaper-elevation6-28"
  role="alert"
>
  <div
    class="MuiAlert-icon-16"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-47 MuiSvgIcon-fontSizeInherit-53"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    class="MuiAlert-message-17"
  >
    Can't load the movie list from the server!
  </div>
</div>
`;

exports[`MovieList Storybook stories snapshot tests Story primary 1`] = `
<ul
  class="MuiList-root-3 makeStyles-root-0 MuiList-padding-4"
>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-348"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien (8.1)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-126889"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien: Covenant (5.9)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-8077"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        AlienÂ³ (6.3)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Action
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-593035"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Warfare (3.9)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Action
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Thriller
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-8078"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Resurrection (6.1)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Action
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-668343"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Outbreak (4.1)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-395"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        AVP: Alien vs. Predator (5.8)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Adventure
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Action
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-561859"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Overlords (4.6)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Action
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-258193"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Abduction (5.1)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Thriller
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-25642"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Ben 10 Alien Swarm (6.2)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Adventure
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-312526"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Outpost (4.7)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Thriller
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Action
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-527014"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Psychosis (4.7)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-22076"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Hunter (5.2)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Action
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Adventure
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-13954"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Raiders (5.3)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Thriller
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-588209"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien: Harvest (6.5)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-460962"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Convergence (3.7)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Action
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
  <div
    aria-disabled="false"
    class="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-gutters-14 MuiListItem-button-15"
    data-testid="movie-548257"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root-21"
    >
      <span
        class="MuiTypography-root-27 MuiListItemText-primary-25 MuiTypography-body1-29 MuiTypography-displayBlock-56"
      >
        Alien Predator (4.4)
      </span>
    </div>
    <div
      class="makeStyles-genreContainer-1"
    >
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Action
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Science Fiction
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Adventure
        </span>
      </div>
      <div
        class="MuiChip-root-57 MuiChip-sizeSmall-58"
      >
        <span
          class="MuiChip-label-79 MuiChip-labelSmall-80"
        >
          Horror
        </span>
      </div>
    </div>
    <span
      class="MuiTouchRipple-root-94"
    />
  </div>
  <li
    class="MuiDivider-root-87"
    role="separator"
  />
</ul>
`;
